File: Data Structures & Algorithms/max-consecutive-ones/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums) {

        let max = 0;
        let left = 0;
        let right = 0;

        while(right <= nums.length) {

            if(nums[right] === 1) {
                left = right;
                let currMax = 0;
                while(right < nums.length && nums[right] === 1) {
                    currMax += 1;
                    max = Math.max(max, currMax);
                    right += 1;
                }
            }

            right += 1;

        }

        return max;

    }
}

File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        i = 0

        while i < len(nums):
            if nums[i] == val:
                j = i + 1
                if j >= len(nums):
                    return i
                while j < len(nums):
                    if nums[j] != val:
                        nums[i] = nums[j]
                        nums[j] = val
                        break
                    j += 1
                if j == len(nums):
                    return i
            i += 1
        
        return i
